### DISCUSSION TITLE
MAUI Operating system

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/maui#8508] [Ideas]
Just like chromebooks, but you can use web / dotNet technologies..
We can do that already because I can access my camera I can check my screen dimensions etc using MAUI.Blazor, I can write applications that has animation just as easy as I would with React while it is blazingly responsive, yet if I want I can also put bytes directly on the webview and make a video stream player out of C# codes I write or a nuget package

**Comment by REscobar** [comment]
I don't want to be sarcastic, but Windows 10 and 11 are already available on mobile form factors such as laptops

**Reply by Xyncgas** [reply]
oh... I also don't want to be sarcastic, but this is virus free

**Reply by Xyncgas** [reply]
It can also use linux or any os under the hood, with a industry api specification of how to expose the device capabilities, so any managed engine can pick it up, give it to IDE, from there you can use any language to write applications for any platform that's gonna look mostly the same due to unified UI via brow rendering.

OS : CommonRT & Unified UI : Any Language IDE & Compatible IL, gives a way for parallel ecosystems for people to use Microsoft or other companies' technology to write once for all platform without having to use a certain language for a certain platform

Users free to choose what OS they want to install, dev can choose what language they want to talk to the computer, market becomes open for competitions of OS and languages and we can even see packages written in different languages to be available in the IDE if they have a compiler that compiles to the target CommonRT, or we can see different companies refuse to open source or officially support compiling their language for a competitor's virtual machine, I think it should be this way, and it should be legislated to be this way, because users can choose anything, and private business can be a private business, but the walled garden stays out of the hardware that people own

**Comment by Theoistic** [comment]
In that case I would not use maui, look at building the linux kernel with a QT webview .. then you can host asp.net blazor locally and use the webview as the OS UI with linux powering the system..

Maui has features and all the bells and whistles when you first look at it, but there are a lot of limitations, the fact that the Windows part is forced with WinUI is kinda stealing apples trademark "dont let the developers have the power" - apple

the WebView OS approch has been done often, look at ElementOS ..

ps. dont say things are virus free.
